File: features/pages/login_page.py
from selenium.webdriver.chrome.webdriver import WebDriver


class LoginHomePage:

    def __init__(self, driver: WebDriver):
        self.driver = driver

    def url(self):
        self.driver.get("http://127.0.0.1:5500/login.html")
        return self.driver.maximize_window()

    def username_input(self):
        return self.driver.find_element_by_id("usernameInput")

    def password_input(self):
        return self.driver.find_element_by_id("passwordInput")

    def login_button(self):
        return self.driver.find_element_by_id("loginButton")

    def customer_login(self):
        self.username_input().send_keys("jose")
        self.password_input().send_keys("12345")
        self.login_button().click()

File: JavaAPI/src/test/java/pages/LoginPage.java
package pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class LoginPage {
    @FindBy(id = "usernameInput")
    private WebElement username;

    @FindBy(id = "passwordInput")
    private WebElement password;

    @FindBy(id = "loginButton")
    private WebElement loginButton;

    @FindBy(xpath = "/html/body/div[2]/div[1]/p")
    private WebElement firstProduct;

    public LoginPage(WebDriver driver) {
        PageFactory.initElements(driver, this);
    }

    public WebElement inputUsername() {
        return this.username;
    }

    public WebElement inputPassword() {
        return this.password;
    }

    public void clickLogin() {
        this.loginButton.click();
    }

    public WebElement getPopularProduct() {
        return this.firstProduct;
    }
}
